src/util.h

```diff
@@ -128,6 +151,7 @@ class ArgsManager
     std::map<std::string, std::vector<std::string>> m_config_args;
     std::string m_network;
     std::set<std::string> m_network_only_args;
+    std::map<std::pair<OptionsCategory, std::string>, std::pair<std::string, bool>> m_available_args;
```

Comment:
I think @promag's suggestion is better; it's more readable (struct with named arguments), and should be more efficient when looking up arguments.

Also, it can avoid duplicating the Arg objects by using `std::map<OptionsCategory, std::map<std::string, const Arg*>> m_args_by_category` instead.

Reply:
@sipa bit late for that now though. The commit with that change was merged in a different PR with just that commit.

Reply:
@achow101 I disagree - the current data structure were sufficient for the use case in that PR (producing the help message), plus that PR did the bulk of the needs-frequent-rebase changes (the `gArgs.AddArg` calls everywhere).

I believe that a better data structure is possible for the new use case this PR introduces, and will probably simplify the PR too.

Reply:
I've made some changes to the structure that are similar to what was suggested. Now I am using a single map which has a map nested inside. The key for the outer map is the category, and the value is a map for all of the args in that category. I am also using a struct to hold the help text and debug_only bool.